technic_chests/register.lua

```diff
@@ -28,59 +28,61 @@ local node_groups_no_inv = {
 	not_in_creative_inventory = 1,
 }
 
-local function get_tiles(name, data, color)
-	local tiles = {
-		"technic_"..name.."_chest_top.png"..tube_entry,
-		"technic_"..name.."_chest_top.png"..tube_entry,
-		"technic_"..name.."_chest_side.png"..tube_entry,
-		"technic_"..name.."_chest_side.png"..tube_entry,
-		"technic_"..name.."_chest_side.png"..tube_entry,
-		"technic_"..name.."_chest_front.png"
+local function get_tiles(data, color)
+	local tiles = data.tiles and table.copy(data.tiles) or {
+		data.texture_prefix.."_top.png"..tube_entry,
+		data.texture_prefix.."_top.png"..tube_entry,
+		data.texture_prefix.."_side.png"..tube_entry,
+		data.texture_prefix.."_side.png"..tube_entry,
+		data.texture_prefix.."_side.png"..tube_entry,
+		data.texture_prefix.."_front.png"
 	}
 	if data.color and color then
 		tiles[6] = tiles[6].."^technic_chest_overlay_"..technic.chests.colors[color][1]..".png"
 	end
 	if data.locked then
-		tiles[6] = tiles[6].."^technic_"..name.."_chest_lock_overlay.png"
+		tiles[6] = tiles[6].."^"..data.texture_prefix.."_lock_overlay.png"
 	elseif data.protected then
 		tiles[6] = tiles[6]..protector_overlay
 	end
 	return tiles
 end
 
-function technic.chests.register_chest(name, data)
-	local lname = name:lower()
-	name = S(name)
+function technic.chests.register_chest(nodename, data)
+	assert(data.tiles or data.texture_prefix, "technic.chests.register_chest: tiles or texture_prefix required")
+	assert(data.description, "technic.chests.register_chest: description required")
+	local colon, modname, name = nodename:match("(:?)([^:]+):(.+)")
+	nodename = modname..":"..name
```

Comment:
This seems unnecessary, why not just let `minetest.register_node` handle the colon check?

Reply:
Because someone said that drops and so on don't like colons.

Reply:
Ah, right... In that case it could be simplified:
```suggestion
	nodename = name:gsub("^:", "")
```
changing the other lines to:
```
minetest.register_node(":"..nodename, def)
```

Reply:
True that should work now just fine 👍 , had also texture prefixes generated there earlier so that's why there was full breakdown first.
Having default texture prefix generated there however got complicated (would require more changes and texture renaming) for builtin chests so moved it away.

Reply:
Actually it wont be enough because it will lose colon for registration if one is supplied but still right direction.

Reply:
Updated to leave modname out of it and capture just colon + full nodename 5be57fa